cmd/pctl/get.go

```diff
@@ -0,0 +1,269 @@
+	}
+	catalogName, profileName := parts[0], parts[1]
+	manager := &catalog.Manager{}
+	profile, err := manager.Show(catalogClient, catalogName, profileName, version)
+	if err != nil {
+		return err
+	}
+
+	return formatCatlogProfilesOutput(profile, outFormat)
+}
+
+func profilesDataFunc(profiles []profilesv1.ProfileCatalogEntry) func() interface{} {
+	return func() interface{} {
+		tc := formatter.TableContents{
+			Headers: []string{"Catalog/Profile", "Version", "Description"},
+		}
+		for _, profile := range profiles {
+			tc.Data = append(tc.Data, []string{
+				fmt.Sprintf("%s/%s", profile.CatalogSource, profile.Name),
+				profilesv1.GetVersionFromTag(profile.Tag),
+				profile.Description,
+			})
+		}
+		return tc
+	}
+}
+
+func installedProfilesDataFunc(data []catalog.ProfileData) func() interface{} {
+	return func() interface{} {
+		tc := formatter.TableContents{
+			Headers: []string{"Namespace", "Name", "Source", "Available Updates"},
+		}
+		for _, d := range data {
+			source := fmt.Sprintf("%s/%s/%s", d.Profile.Catalog, d.Profile.Profile, d.Profile.Version)
+			if d.Profile.Catalog == "-" {
+				source = fmt.Sprintf("%s:%s:%s", d.Profile.URL, d.Profile.Branch, d.Profile.Path)
+			}
+			tc.Data = append(tc.Data, []string{
+				d.Profile.Namespace,
+				d.Profile.Name,
+				source,
+				strings.Join(d.AvailableVersionUpdates, ","),
+			})
+		}
+		return tc
+	}
+}
+
+func profileWithVersionDataFunc(profile profilesv1.ProfileCatalogEntry) func() interface{} {
+	return func() interface{} {
+		return formatter.TableContents{
+			Data: [][]string{
+				{"Catalog", profile.CatalogSource},
+				{"Name", profile.Name},
+				{"Version", profilesv1.GetVersionFromTag(profile.Tag)},
+				{"Description", profile.Description},
+				{"URL", profile.URL},
+				{"Maintainer", profile.Maintainer},
+				{"Prerequisites", strings.Join(profile.Prerequisites, ", ")},
+			},
+		}
+	}
+}
+
+func formatOutput(profiles []profilesv1.ProfileCatalogEntry, outFormat string) error {
+	if len(profiles) == 0 {
+		fmt.Println("No profiles found")
+		return nil
+	}
+
+	var f formatter.Formatter
+	f = formatter.NewTableFormatter()
+	getter := profilesDataFunc(profiles)
+
+	if outFormat == "json" {
+		f = formatter.NewJSONFormatter()
+		getter = func() interface{} { return profiles }
+	}
+
+	out, err := f.Format(getter)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("PACKAGE CATALOG")
+	fmt.Println(out)
+	return nil
+}
+
+func formatInstalledProfilesOutput(data []catalog.ProfileData, outFormat string) error {
+	if len(data) == 0 {
+		fmt.Println("No profiles installed")
+		return nil
+	}
+
+	var f formatter.Formatter
+	f = formatter.NewTableFormatter()
+	getter := installedProfilesDataFunc(data)
+
+	if outFormat == "json" {
+		f = formatter.NewJSONFormatter()
+		getter = func() interface{} { return data }
+	}
+
+	out, err := f.Format(getter)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("INSTALLED PACKAGES")
```

Comment:
does this break the `--output json` options?

Reply:
so this output is not included inside the json object. for eg.,
```
✗ $ pctl get -o json
INSTALLED PACKAGES
[
  {
    "Profile": {
      "Name": "pprof",
      "Namespace": "7959ad7d-fa40-463a-91f8-550ff7adb9cc",
      "Version": "-",
      "Profile": "-",
      "Catalog": "-",
      "Branch": "main",
      "Path": "weaveworks-nginx",
      "URL": "https://github.com/weaveworks/profiles-examples"
    },
    "AvailableVersionUpdates": [
      "-"
    ]
  },
  {
    "Profile": {
      "Name": "pprof",
      "Namespace": "d90fee4c-2810-4988-b517-873a9b9b93c3",
      "Version": "-",
      "Profile": "-",
      "Catalog": "-",
      "Branch": "main",
      "Path": "weaveworks-nginx",
      "URL": "https://github.com/weaveworks/profiles-examples"
    },
    "AvailableVersionUpdates": [
      "-"
    ]
  },
  {
    "Profile": {
      "Name": "bitnami-profile",
      "Namespace": "default",
      "Version": "v0.0.1",
      "Profile": "bitnami-nginx",
      "Catalog": "nginx-catalog",
      "Branch": "main",
      "Path": "-",
      "URL": "https://github.com/weaveworks/profiles-examples"
    },
    "AvailableVersionUpdates": [
      "-"
    ]
  }
]
PACKAGE CATALOG
[
  {
    "tag": "weaveworks-nginx/v0.1.0",
    "catalogSource": "nginx-catalog",
    "url": "https://github.com/weaveworks/profiles-examples",
    "name": "weaveworks-nginx",
    "description": "This installs nginx.",
    "maintainer": "weaveworks (https://github.com/weaveworks/profiles)",
    "prerequisites": [
      "Kubernetes 1.18+"
    ]
  },
  {
    "tag": "weaveworks-nginx/v0.1.1",
    "catalogSource": "nginx-catalog",
    "url": "https://github.com/weaveworks/profiles-examples",
    "name": "weaveworks-nginx",
    "description": "This installs nginx.",
    "maintainer": "weaveworks (https://github.com/weaveworks/profiles)",
    "prerequisites": [
      "Kubernetes 1.18+"
    ]
  },
  {
    "tag": "bitnami-nginx/v0.0.1",
    "catalogSource": "nginx-catalog",
    "url": "https://github.com/weaveworks/profiles-examples",
    "name": "bitnami-nginx",
    "description": "This installs nginx.",
    "maintainer": "weaveworks (https://github.com/weaveworks/profiles)",
    "prerequisites": [
      "Kubernetes 1.18+"
    ]
  },
  {
    "tag": "v2.0.1",
    "catalogSource": "nginx-catalog",
    "url": "https://github.com/weaveworks/nginx-profile",
    "name": "nginx",
    "description": "This installs nginx.",
    "maintainer": "weaveworks (https://github.com/weaveworks/profiles)",
    "prerequisites": [
      "Kubernetes 1.18+"
    ]
  },
  {
    "catalogSource": "nginx-catalog",
    "name": "some-other-nginx",
    "description": "This installs some other nginx."
  }
]

```
do we want this line to be included inside the json output?

Reply:
I think the goal of `--output json` is that the whole output is valid json, e.g. `pctl get -o json | jq .`. I think anything that makes the output not valid json shouldn't be printed.

Reply:
I have added changes so the JSON output doesn't break